api/swagger.yml

```diff
@@ -1,507 +1,795 @@
-swagger: "2.0"
+openapi: "3.0.0"
 
-consumes:
-  - application/json
 info:
   description: lakeFS HTTP API
   title: lakeFS API
   version: 0.1.0
-produces:
-  - application/json
-schemes:
-  - http
-  - https
-basePath: /api/v1
-
-
-securityDefinitions:
-  basic_auth:
-    type: basic
-  jwt_token:
-    type: apiKey
-    in: header
-    name: X-JWT-Authorization
+
+
+servers:
+  - url: '/api/v1'
+    description: lakeFS server endpoint
+    variables:
+      protocol:
+        enum:
+          - http
+          - https
+        default: http
+      server_address:
+        default: 'localhost:8000'
 
 security:
   - jwt_token: []
   - basic_auth: []
 
-responses:
-  Unauthorized:
-    description: Unauthorized
-    schema:
-      $ref: "#/definitions/error"
-
-definitions:
-
-  pagination:
-    type: object
-    required:
-      - has_more
-    properties:
-      has_more:
-        type: boolean
-      next_offset:
-        type: string
-      results:
-        type: integer
-        minimum: 0
-      max_per_page:
-        type: integer
-        minimum: 0
-
-  repository:
-    type: object
-    properties:
-      id:
-        type: string
-      creation_date:
+components:
+
+  securitySchemes:
+    basic_auth:
+      type: http
+      scheme: basic
+    jwt_token:
+      type: apiKey
+      in: header
+      name: X-JWT-Authorization
+
+  parameters:
+    PaginationAfter:
+      in: query
+      name: after
+      description: return items after this value
+      schema:
+        type: string
+    PaginationAmount:
+      in: query
+      name: amount
+      description: how many items to return
+      schema:
         type: integer
-        format: int64
-      default_branch:
-        example: "master"
-        type: string
-      storage_namespace:
-        type: string
-        description: "Filesystem URI to store the underlying data in (e.g. 's3://my-bucket/some/path/')"
-
-  merge_result:
-    type: object
-    properties:
-      summary:
-        type: object
-        properties:
-          added:
-            type: integer
-          removed:
-            type: integer
-          changed:
-            type: integer
-          conflict:
-            type: integer
-      reference:
-        type: string
-
-  repository_creation:
-    type: object
-    required:
-      - name
-      - storage_namespace
-    properties:
-      name:
-        type: string
-        pattern: '^[a-z0-9][a-z0-9-]{2,62}$'
-      storage_namespace:
-        type: string
-        description: "Filesystem URI to store the underlying data in (e.g. 's3://my-bucket/some/path/')"
-        example: "s3://example-bucket/"
-        pattern: '^(s3|gs|https?|mem|local|transient)://.*$'
-      default_branch:
-        example: "master"
-        type: string
+        minimum: -1
```

Comment:
Can we cap it at minimum 0 and have "missing" to mean "use server default"?

Reply:
Today zero means - I don't want results - just has more indication.
We can change it and have 0 as the server's default, but I think we can do it in a different issue if we all agree that we will lose the capability.